### DISCUSSION TITLE
[Feature Request]: admin channel: setting location

### DISCUSSION THREAD

**Discussion by jranma** [meshtastic/firmware#4285] [Ideas]
### Platform

other

### Description

Please enable remote localization setting using the admin channel. 
Recently I've had an annoying bug with a node that lost its fixed location, for some reason. 
Remote reouters would benefit .

**Comment by garthvh** [comment]
To date we have not allowed nodes to set each other's location for security reasons, and generally spoofing of positions is not a necessary feature with position precision available.

**Reply by jranma** [reply]
I don't get the security argument. If it's done via the admin channel, security would be at the same level as for any other option, wouldn't it? 

One of my routers lost its location after a few days of use, without me knowing why, and it doesn't have GPS (fixed location). whether the location is accurate or not is another matter altogether. 
I had to climb back up to the top of the summit with my laptop in my backpack, a 3-hour walk. Of course, sport is always good for your health, and the weather was excellent! No regrets here, and that's part of the fun... But there should be another solution ;-)

**Reply by Garrisonsan** [reply]
I have the same issue with a router node dumping its fixed coordinates randomly. It seems to be after 30-45 days, then requiring me to either make a special trip to go reprogram it or wait until I'm at that particular site for other independent reasons.

I, too, don't see the risk in being able to remotely set its fixed coordinates over the admin channel.

If I wanted to spoof the location of a node I'm connected to, it's pretty trivial to accomplish that once I've enabled developer mode on Android or quite trivial to do over CLI with some python know-how.

**Reply by garthvh** [reply]
> I don't get the security argument. If it's done via the admin channel, security would be at the same level as for any other option, wouldn't it?

Pretty standard practice for security to avoid scenarios where virtually all the outcomes are negative.  We have a low precision option available.

**Comment by jranma** [comment]
There is a misunderstanding. The idea here is not about spoofing or being more or less precise with location.
Its about being able to set a location remotely for nodes without GPS, as some nodes will lose it for some reason (i experienced twice). Today, when such event happens, the only way to fix it is to gain physical or BT access to the node. Looks like exactly the reason why admin channel has been implemented.

**Reply by garthvh** [reply]
Then open a ticket and work to fix the bug, the fact that there is a bug is not an opportunity to increase position spoofing. Any location not from a GPS is essentially junk data.

**Comment by jranma** [comment]
At this point, you decide to deliberately go off topic: what does this have to do with spoofing? 
I give up. 
Perhaps a developer will be able to judge whether there's anything to be done here.

**Reply by garthvh** [reply]
Your request is for remote position spoofing, I don't see the utility of this feature when we have position precision already.  Adds nothing but risk with very little benefit.

**Comment by joneyes** [comment]
IMO, **_all config_** should be possible for a remote node via the admin channel.
The vulnerabilities imposed by adding an admin channel are the node operators risk to manage and balance.
I'm in the same scenario have a remote node 30 minutes drive from here. I'd sooner be able to make any config change i want via the admin channel rather than an hours round trip plus 30 minutes walk.
If you want it secure, then don't add an admin channel.

**Reply by joneyes** [reply]
PS - i can make channel and Lora mode changes remotely via the admin channel - that has a much bigger risk than being able to change lat/long/alt - node goes offline rather than moving!
